capstone-ui/src/recommendation.js

```diff
@@ -1,103 +1,61 @@
-//   //if dataset is in p1 and p2
-//   //store it in as one
-//   for (var key in dataset[p1]) {
-//     console.log('key: ' + key);
-//     // if (key in dataset[p2]) {
-//     //   existp1p2[key] = 1;
-//     // }
-//     if (dataset[p2].hasOwnProperty(key)) {
-//       existp1p2[key] = 1;
-//     }
-//     if (len(existp1p2) == 0) return 0; //check if it has a data
-//     var sum_of_euclidean_dist = []; //store the  euclidean distance
-
-//     //calculate the euclidean distance
-//     for (item in dataset[p1]) {
-//       if (item in dataset[p2]) {
-//         sum_of_euclidean_dist.push(
-//           Math.pow(dataset[p1][item] - dataset[p2][item], 2)
-//         );
-//       }
-//     }
-//     var sum = 0;
-//     for (var i = 0; i < sum_of_euclidean_dist.length; i++) {
-//       sum += sum_of_euclidean_dist[i]; //calculate the sum of the euclidean
-//     }
-//     //since the sum will be small for familiar user
-//     // and larger for non-familiar user
-//     //we make it exist btwn 0 and 1
-//     var sum_sqrt = 1 / (1 + Math.sqrt(sum));
-//     return sum_sqrt;
-//   }
-// };
-
-// euclidean_score(dataset, "Lisa Rose", "Jack Mathews");
-
-// console.log(euclidean_score);
+import axios from 'axios';
+
+function rank(recommendedUsers) {
+    recommendedUsers.sort(function(a, b){
+        return b.score - a.score;
+    });    
+}
+
+export const getRecommendedUsers = async (username, topGenres, topArtists, postedSongs) => {
+    let allUsers = await axios.get(
+        `http://localhost:3001/user/users`
+    );
+
+    let friends = await axios.get(
+        `http://localhost:3001/user/followers/${username}`
+    );
+
+    allUsers = allUsers.data.map(element => element.username);
+    allUsers = allUsers.filter(element => element != username);
+    friends = friends.data;
+    // skip people who are already friends - can't recommend them
+    allUsers = allUsers.filter(x => !friends.includes(x));
+    let result = [];
+
+    console.log('getting recommended users');
+    console.log(allUsers);
+    for (let i = 0; i < allUsers.length; i++) {
+        let score = 0;
+        let friendUsername = allUsers[i];
+        let friendArtists = await axios.get(`http://localhost:3001/user/topartists/${friendUsername}`);
+        let friendGenres = await axios.get(`http://localhost:3001/user/topgenres/${friendUsername}`);
+
+        console.log('mine');
+        console.log(topArtists);
+        console.log('friend' + friendUsername);
+        console.log(friendArtists.data);
+
+        // weight shared artist preferences more heavily than genres since more rare to like a specific artist
+        for (let j = 0; j < friendArtists.data.length; j++) {
+            let currArtist = friendArtists.data[j];
+            let isInArray = topArtists.some(e => e.value === currArtist.value); 
+            if (isInArray) {
+                score += 5;
+            }
+        }
+
+        for (let j = 0; j < friendGenres.data.length; j++) {
```

Comment:
could you add a comment to explain this a bit too like above. why is this weighted smaller?